File: scripts/final-fix.py
#!/usr/bin/env python3
"""
Code.jsの全構文エラーを修正する最終スクリプト
"""

import re
import sys

def fix_code_js(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # 1. 関数定義の戻り値型を削除
    # function name() { type: value; ... } { → function name() {
    content = re.sub(
        r'function\s+(\w+)\s*\(([^)]*)\)\s*\{[^{]+\{',
        r'function \1(\2) {',
        content
    )
    
    # 2. doGet関数のtryを追加
    content = re.sub(
        r'function doGet\(e\) \{(\s*)// eがundefinedの場合',
        r'function doGet(e) {\1try {\1  // eがundefinedの場合',
        content
    )
    
    # 3. doPost関数のtryを追加  
    content = re.sub(
        r'function doPost\(e\) \{(\s*)// eがundefinedの場合',
        r'function doPost(e) {\1try {\1  // eがundefinedの場合',
        content
    )
    
    # 4. include関数のtryを追加
    content = re.sub(
        r'function include\(filename\) \{(\s*)return HtmlService',
        r'function include(filename) {\1try {\1  return HtmlService',
        content
    )
    
    # 5. 短縮プロパティを展開
    content = content.replace(
        'JSON.stringify({ success, data })',
        'JSON.stringify({ success: true, data: result })'
    )
    content = content.replace(
        'JSON.stringify({\n        success,\n        error,\n      })',
        'JSON.stringify({\n        success: false,\n        error: errorMessage,\n      })'
    )
    
    # 6. default) を default: に修正
    content = re.sub(r'\bdefault\)', 'default:', content)
    
    # 7. HTMLタグを修正
    content = content.replace(
        'エラーが発生しました</h1>',
        '<h1>エラーが発生しました</h1>'
    )
    content = content.replace(
        '${errorMessage}</p>',
        '<p>${errorMessage}</p>'
    )
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)
    
    print('✅ Code.jsの修正が完了しました')

if __name__ == '__main__':
    import os
    script_dir = os.path.dirname(os.path.abspath(__file__))
    code_js_path = os.path.join(script_dir, '../dist/Code.js')
    fix_code_js(code_js_path)

File: scripts/final-fix.js
/**
 * 最終修正: 全てのシンタックスエラーを修正
 */

const fs = require('fs');
const path = require('path');

const distPath = path.join(__dirname, '../dist/Code.js');

console.log('🔧 最終修正を開始...');

let content = fs.readFileSync(distPath, 'utf8');

// 1. 重複したreturn文を削除
content = content.replace(/return \{ success: true, data: data \};: true, data \};/g, 'return { success: true, data: data };');
content = content.replace(/return \{ success: true, data: data \};\s*success: true, data: data \};/g, 'return { success: true, data: data };');

// 2. 壊れたオブジェクトリテラルを修正
// success, data ) → success: true, data: data })
content = content.replace(/\{ success, data \)/g, '{ success: true, data: data })');
content = content.replace(/\{ success, message \)/g, '{ success: false, message: \'Error\' })');

// 3. 型注釈の残骸を全て削除
content = content.replace(/: boolean/g, '');
content = content.replace(/: string/g, '');
content = content.replace(/: number/g, '');
content = content.replace(/: object/g, '');
content = content.replace(/: any/g, '');
content = content.replace(/\?: /g, '');

// 4. 空白の多い行を整理
content = content.replace(/\n\n\n+/g, '\n\n');

// 5. 保存
fs.writeFileSync(distPath, content);

console.log('✅ 最終修正完了');
